pandas/core/generic.py

```diff
@@ -1408,16 +1408,18 @@ def __neg__(self):
 
     def __pos__(self):
         values = self._values
-        if is_bool_dtype(values):
-            arr = values
-        elif (
-            is_numeric_dtype(values)
+        if (
+            is_bool_dtype(values)
+            or is_numeric_dtype(values)
             or is_timedelta64_dtype(values)
             or is_object_dtype(values)
         ):
-            arr = operator.pos(values)
+            arr = values
         else:
-            raise TypeError(f"Unary plus expects numeric dtype, not {values.dtype}")
+            raise TypeError(
+                "Unary plus expects bool, numeric, timedelta, "
+                f"or object dtype, not {values.dtype}"
```

Comment:
are we actually testing this message?

Reply:
Yes, the updated test is in /frame/test_operators.py